PWGCF/Flow/Tasks/flowJSPCAnalysisTask.cxx

```diff
@@ -0,0 +1,207 @@
+  } cfgEventCuts;
+
+  // Set the access to the CCDB for the NUA/NUE weights.
+  struct : ConfigurableGroup {
+    Configurable<bool> cfgUseCCDB{"cfgUseCCDB", true, "Use CCDB for NUA/NUE corrections."};
+    Configurable<std::string> cfgURL{"cfgURL", "http://alice-ccdb.cern.ch",
+                                     "Address of the CCDB to get the NUA/NUE."};
+    Configurable<int64_t> cfgTime{"ccdb-no-later-than",
+                                  std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(),
+                                  "Latest acceptable timestamp of creation for the object."};
+  } cfgCCDB;
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+
+  // // Filters to be applied to the received data.
+  // // The analysis assumes the data has been subjected to a QA of its selection,
+  // // and thus only the final distributions of the data for analysis are saved.
+  Filter collFilter = (nabs(aod::collision::posZ) < cfgEventCuts.cfgZvtxMax);
+  Filter trackFilter = (aod::track::pt > cfgTrackCuts.cfgPtMin) && (aod::track::pt < cfgTrackCuts.cfgPtMax) && (nabs(aod::track::eta) < cfgTrackCuts.cfgEtaMax);
+
+  void init(InitContext const&)
+  {
+    // Add histomanager here
+    spcAnalysis.SetHistRegistry(&SPCHistograms);
+    spcAnalysis.CreateHistos();
+
+    SPCobservables.SetSPCObservables(cfgWhichSPC);
+    spcAnalysis.SetFullCorrSet(SPCobservables.harmonicArray);
+
+    histManager.SetHistRegistryQA(&qaHistRegistry);
+    histManager.SetDebugLog(false);
+    histManager.CreateHistQA();
+
+    /////////////////////
+
+    // Analysis class initialization
+    // blaa blaa spcAnalysis.CreateOutputObjects()
+
+    ccdb->setURL(cfgCCDB.cfgURL);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    ccdb->setCreatedNotAfter(cfgCCDB.cfgTime.value);
+
+    // Add CCDB access here
+  }
+
+  void process(soa::Filtered<MyCollisions>::iterator const& coll, soa::Filtered<MyTracks> const& tracks)
+  {
+    if (tracks.size() < cfgEventCuts.cfgMultMin)
+      return;
+    // Int_t nTracks = tracks.size();
+
+    float cent = -1.;
+    switch (cfgEventCuts.cfgCentEst) {
+      case FT0M:
+        cent = coll.centFT0M();
+        break;
+      case FT0A:
+        cent = coll.centFT0A();
+        break;
+      case FT0C:
+        cent = coll.centFT0C();
+        break;
+      case FDDM:
+        cent = coll.centFDDM();
+        break;
+      case NTPV:
+        cent = coll.centNTPV();
+        break;
+    }
+    if (cent < 0. || cent > 70.) {
+      return;
+    }
+    Int_t cBin = histManager.GetCentBin(cent);
+    // printf("Centrality: %.2f,\tBin: %d\n", cent, cBin);
+    SPCHistograms.fill(HIST("FullCentrality"), cent);
+    vector<float> trackPhi;
+    vector<float> trackWeight;
+    int nTracks = 0;
+
+    for (auto& track : tracks) {
+      if (cfgFillQA)
+        histManager.FillTrackQA<1>(track, cBin, coll.posZ());
+      trackPhi.push_back(track.phi());
```

Comment:
In here the phi column of the tracks table is replicated in memory and this is not efficient
Please pass the actual table instead to the involved functions 'templating' them if it were needed

Reply:
You are correct, it's inefficient. This has been now corrected for the future task.